extensions/common/gcp/gcp-core/src/test/java/org/eclipse/dataspaceconnector/gcp/core/iam/IamServiceImplTest.java

```diff
@@ -0,0 +1,156 @@
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LCC - Initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.gcp.core.iam;
+
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.ApiExceptionFactory;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.cloud.iam.admin.v1.IAMClient;
+import com.google.cloud.iam.credentials.v1.GenerateAccessTokenRequest;
+import com.google.cloud.iam.credentials.v1.GenerateAccessTokenResponse;
+import com.google.cloud.iam.credentials.v1.IamCredentialsClient;
+import com.google.iam.admin.v1.CreateServiceAccountRequest;
+import com.google.iam.admin.v1.ServiceAccount;
+import org.eclipse.dataspaceconnector.gcp.core.common.GcpException;
+import org.eclipse.dataspaceconnector.gcp.core.common.ServiceAccountWrapper;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.AdditionalMatchers.and;
+import static org.mockito.Answers.RETURNS_SMART_NULLS;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.argThat;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class IamServiceImplTest {
+
+    private final String projectId = "test-project-Id";
+    private final String serviceAccountName = "test-service-account";
+    private final String serviceAccountEmail = String.format("%s@%s.iam.gserviceaccount.com", serviceAccountName, projectId);
+    private final String serviceAccountDescription = "service-account-description";
+    private IamService iamApi;
+    private IAMClient iamClient;
+    private IamCredentialsClient iamCredentialsClient;
+    private ServiceAccountWrapper testServiceAccount;
+
+    private static ApiException apiExceptionWithStatusCode(StatusCode.Code code) {
+        return ApiExceptionFactory.createException(
+                new Exception(), new StatusCode() {
+                    @Override
+                    public Code getCode() {
+                        return code;
+                    }
+
+                    @Override
+                    public Object getTransportCode() {
+                        return null;
+                    }
+                }, false);
+    }
+
+    @BeforeEach
+    void setUp() {
+        var monitor = Mockito.mock(Monitor.class);
+        iamClient = Mockito.mock(IAMClient.class, RETURNS_SMART_NULLS);
```

Comment:
Are these `RETURNS_SMART_NULLS` needed? tests will pass also without it, I will suggest to remove them where not needed.